src/main/java/net/sf/jabref/model/database/EntrySorter.java

```diff
@@ -90,8 +90,10 @@ public void databaseChanged(DatabaseChangeEvent e) {
             int pos;
             switch (e.getType()) {
             case ADDED_ENTRY:
-                pos = -Collections.binarySearch(set, e.getEntry(), comp) - 1;
-                set.add(pos, e.getEntry());
+                pos = Collections.binarySearch(set, e.getEntry(), comp);
```

Comment:
Thank you for your explanation. Now it makes kind of sense. 
A minimal test just would call databaseChanged. A more elaborate way is to register the EntrySorter to a test database and then remove or add items there. Both tests should be more or less straightforward.